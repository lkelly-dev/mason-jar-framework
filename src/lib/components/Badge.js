import React from 'react';
import { cx } from 'emotion';
import MODIFIERS from './Modifiers';

const Badge = props => {
  const { children, color, className, ...rest } = props;
  const badgeColor = MODIFIERS.COLOR[color] ? `badge-${color}` : null;
  const badgeClass = cx('badge', badgeColor, className);
  return (
    <span className={badgeClass} {...rest}>
      {children}
    </span>
  );
};

export { Badge };
