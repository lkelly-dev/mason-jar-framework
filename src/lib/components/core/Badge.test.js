import React from 'react';
import { shallow } from 'enzyme';
import { Badge } from "./Badge";

describe('Badge', () => {
  it('matches snapshot', () => {
    const component = shallow(<Badge></Badge>);
    expect(component).toMatchSnapshot();
  })
  it('has correct props', () => {
    const component = shallow(<Badge color="danger" className="neato" eggman="walrus">A</Badge>);
    expect(component.prop("className")).toBe("badge badge-danger neato");
    expect(component.prop('children')).toBe('A');
    expect(component.prop('eggman')).toBe('walrus');
  })
});
