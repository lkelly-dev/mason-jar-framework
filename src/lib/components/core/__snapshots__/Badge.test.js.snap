// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Badge matches snapshot 1`] = `
<span
  className="badge"
/>
`;

exports[`Badge matches snapshot 1`] = `
<span
  className="badge"
/>
`;
